File: Exercism.io/python/sublist/sublist.py

SUBLIST   = 1
SUPERLIST = 2
EQUAL     = 3
UNEQUAL   = 4

def check_lists(a, b):
    if a == b:           return EQUAL
    if is_sublist(a, b): return SUBLIST
    if is_sublist(b, a): return SUPERLIST
    return UNEQUAL

def is_sublist(a, b):
    start, end = 0, len(a)
    while end <= len(b):
        if a == b[start:end]: return True
        start, end = start+1, end+1
    return False

File: Exercism.io/elixir/sublist/sublist.exs
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """
  def compare(a, b) do
    cond do
       a === b      -> :equal
       is_sub?(a,b) -> :sublist
       is_sub?(b,a) -> :superlist
       true         -> :unequal
    end
  end

  defp is_sub?(a, b) do
    cond do
      length(a) > length(b)         -> false
      Enum.take(b, length(a)) === a -> true
      true                          -> is_sub?(a, tl(b))
    end
  end

end
